css/src/components/hero.scss

```diff
@@ -0,0 +1,153 @@
+$hero-padding-xs: $layout-4 !default;
+$hero-padding-sm: $layout-5 !default;
+$hero-padding-default: $layout-6 !default;
+$hero-padding-lg: $layout-7 !default;
+
+$hero-min-height-xs: 200px !default;
+$hero-min-height-sm: 300px !default;
+$hero-min-height-default: 400px !default;
+$hero-min-height-lg: 500px !default;
+
+$hero-content-width: $reading-max-width !default;
+$hero-content-fluid-width: 45% !default;
+$hero-image-width: calc((100% - 2 * $layout-gap) * 0.55) !default;
+$hero-image-width-widescreen: calc((100% - 2 * $layout-widescreen-gap) * 0.55) !default;
+
+.hero {
+	@extend %layout-gap;
+
+	display: flex;
+	position: relative;
+	flex-direction: column;
+	align-items: stretch;
+	justify-content: space-between;
+
+	.hero-content {
+		z-index: $zindex-active;
+		padding-block: $hero-padding-sm;
+
+		@include desktop {
+			width: $hero-content-fluid-width;
+			min-height: $hero-min-height-default;
+			padding-block: $hero-padding-default;
+			padding-inline-end: $hero-padding-sm;
+		}
+	}
+
+	&:not(.hero-image) {
+		.hero-content {
+			@include desktop {
+				width: 100%;
+				max-width: $hero-content-width;
+			}
+		}
+	}
+
+	@include desktop {
+		&.hero-xs {
+			.hero-content {
+				min-height: $hero-min-height-xs;
+				padding-block: $hero-padding-xs;
+			}
+		}
+
+		&.hero-sm {
+			.hero-content {
+				min-height: $hero-min-height-sm;
+				padding-block: $hero-padding-sm;
+			}
+		}
+
+		&.hero-lg {
+			.hero-content {
+				min-height: $hero-min-height-lg;
+				padding-block: $hero-padding-lg;
+			}
+		}
+	}
+
+	.hero-card {
+		position: relative;
+		padding: $spacer-5;
+		border: none;
+		background-color: transparent;
+		z-index: 1;
+
+		@include tablet {
+			padding: $spacer-10;
+		}
+
+		@include desktop {
+			border-right: 1px solid $border;
+			border-bottom: 1px solid $border;
+			border-left: 1px solid $border;
+			border-bottom-left-radius: $border-radius-lg;
+			border-bottom-right-radius: $border-radius-lg;
+		}
+	}
+
+	.hero-background {
+		position: absolute;
+		top: 0;
+		right: 0;
+		bottom: 0;
+		left: 0;
+		width: 100%;
+		height: 100%;
+		overflow: hidden;
+
+		video,
+		img {
+			width: 100%;
+			height: 100%;
+			object-fit: cover;
+		}
+	}
+}
+
+.hero-image {
+	&::before {
```

Comment:
reduce level of nesting by doing `.hero-image::before`?

Reply:
Looking at the example, it seems like `hero-image` is meant as a subclass, so I nested it within `.hero`.